### PR TITLE
chore: improve fixture for compose.test (add strict types)

### PR DESCRIPTION
**Pull request by peter-sanderson**
Part of https://github.com/trezor/trezor-suite/issues/10765

### CODE REVIEW

File: packages/utxo-lib/tests/compose.test.ts [hunk 1]
```diff
@@ -1,27 +1,24 @@
-import { composeTx } from '../src';
+import { composeTx } from '../src/compose';
```
Comment (Lemonexe) [inline packages/utxo-lib/tests/compose.test.ts:1]: Nit: seems like unnecessary change?

File: packages/utxo-lib/tests/compose.test.ts [hunk 2]
```diff
@@ -1,27 +1,24 @@
...
 
 import { verifyTxBytes } from './compose.utils';
 import { composeTxFixture } from './__fixtures__/compose';
 import { fixturesCrossCheck } from './__fixtures__/compose.crosscheck';
 
-// keyof typeof NETWORKS;
-const getNetwork = (name?: string) =>
-    // @ts-expect-error expression of type string can't be used to index type
-    typeof name === 'string' && NETWORKS[name] ? NETWORKS[name] : NETWORKS.bitcoin;
-
```
Comment (Lemonexe) [inline packages/utxo-lib/tests/compose.test.ts:12]: Good riddance :smile: :wave:

File: packages/utxo-lib/tests/compose.test.ts [hunk 3]
```diff
@@ -1,27 +1,24 @@
...
+        const result = { ...f.result };
 
         it(f.description, () => {
-            const tx = composeTx(request as any);
+            const tx = composeTx(request);
             expect(tx).toEqual(result);
 
-            if (tx.type === 'final') {
-                verifyTxBytes(tx, f.request.txType as any, network);
+            expect(f.request.txType).not.toEqual('p2wsh');
```
Comment (Lemonexe) [inline packages/utxo-lib/tests/compose.test.ts:18]: This could imho do with a short comment: why we are explicitly testing that the _**fixtures**_ don't contain a `p2wsh` type transaction? 
I can see that `verifyTxBytes` does not accept a `p2wsh` type transaction, _whatever that is_, but it's not apparent until you play around with the code.

File: packages/utxo-lib/tests/compose.test.ts [hunk 4]
```diff
@@ -36,24 +33,27 @@ describe('composeTx addresses cross-check', () => {
...
             // skip test for each addressType if there is nothing to replace (example: 7 inputs test)
-            const offset = f.request.outputs.find(o => o.address === 'replace-me')
+            const offset = f.request.outputs.find(o => 'address' in o && o.address === 'replace-me')
                 ? addrKeys.length
                 : 1;
 
             addrKeys.slice(0, offset).forEach(addressType => {
                 const key = `${txType}-${addressType}` as keyof typeof f.result;
-                it(`${key} ${f.description}`, () => {
+                it(`${String(key)} ${f.description}`, () => {
```
Comment (Lemonexe) [inline packages/utxo-lib/tests/compose.test.ts:56]: this seems useless?

File: packages/utxo-lib/tests/__fixtures__/compose.crosscheck.ts [hunk 5]
```diff
@@ -1,6 +1,23 @@
...
+import { CoinSelectPaymentType } from '../../src';
 
-export const fixturesCrossCheck = [
+type AnyComposeRequest = ComposeRequest<ComposeInput, ComposeOutput, ComposeChangeAddress>;
+
+type Fixture = {
+    description: string;
+    request: Omit<AnyComposeRequest, 'network'> & {
+        network?: AnyComposeRequest['network'];
+    };
```
Comment (Lemonexe) [inline packages/utxo-lib/tests/__fixtures__/compose.crosscheck.ts:16]: IMHO we should give some thought to explore https://www.npmjs.com/package/@total-typescript/shoehorn 

Not the scope of this PR ofc, this is ok :slightly_smiling_face: 

Also, don't we have some type generic util for Nullable fields on object? Can't find it, so maybe I'm wrong.

File: packages/utxo-lib/tests/__fixtures__/compose.ts [hunk 6]
```diff
@@ -958,7 +983,7 @@ export const composeTxFixture = [
         description: 'builds a simple tx without change (cashaddr)',
         request: {
             changeAddress: { address: 'bitcoincash:qzppkat2v7xu9fr3yeuqdnggjqqltrs7pcg8swvhl0' },
-            network: 'bitcoincash',
+            network: bitcoincash,
```
Comment (Lemonexe) [inline packages/utxo-lib/tests/__fixtures__/compose.ts:986]: This was literally wrong, thx :+1: :+1:

File: packages/utxo-lib/tests/__fixtures__/compose.ts [hunk 7]
```diff
@@ -39,7 +63,7 @@ export const composeTxFixture = [
                     amount: '100000',
                     type: 'payment',
                     customField: 'prove that payment output is generic',
-                },
+                } as AnyComposeFinalResult['outputs'][number], // hack for `customField`
```
Comment (Lemonexe) [inline packages/utxo-lib/tests/__fixtures__/compose.ts:66]: Why don't you declare the same thing for `Fixture.result` as for `Fixture.request` if you want to support `customField`?
Boilerplate is quicker to read than inconsistent approaches.

**Review by Lemonexe — APPROVED** [review]
Thanks a lot! It's a great code improvement, makes the tests less hacky and magic.

I do have some comments for codestyle or TS.

File: packages/utxo-lib/tests/compose.test.ts (hunk 1 above)
Comment (peter-sanderson) [reply packages/utxo-lib/tests/compose.test.ts:1]: I think that we shall not import in test from barrel index files, but rather from the file we are directly testing.


Barrel files just define the outer API of the package, and tests are internal stuff of that package.

File: packages/utxo-lib/tests/compose.test.ts (hunk 3 above)
Comment (peter-sanderson) [reply packages/utxo-lib/tests/compose.test.ts:18]: TBH I just tweak it to pass the checks. This is mystery to me and I dont understand why this is the case.

It was out-of-scope for me.

Comment (Lemonexe) [reply packages/utxo-lib/tests/compose.test.ts:18]: Oh that's fine :ok_hand:   I was just wondering if you have more insight :slightly_smiling_face:

File: packages/utxo-lib/tests/compose.test.ts (hunk 4 above)
Comment (peter-sanderson) [reply packages/utxo-lib/tests/compose.test.ts:56]: I got an error (from Idea I guess) that this is a proper way how to convert `symbol` to string. That just doing 
```
`${...}`
```

may fail. 

Obviously not here, I wanted to get rid of the error.

File: packages/utxo-lib/tests/__fixtures__/compose.ts (hunk 7 above)
Comment (peter-sanderson) [reply packages/utxo-lib/tests/__fixtures__/compose.ts:66]: That doesnt work, you cannot `Omit` a fields as ComposeResult is union type, where that field is not present in all components. :(

Comment (Lemonexe) [reply packages/utxo-lib/tests/__fixtures__/compose.ts:66]: Got it! Did not realise that.
